cpp/src/parquet/types.h

```diff
@@ -588,6 +592,26 @@ inline bool operator!=(const ByteArray& left, const ByteArray& right) {
   return !(left == right);
 }
 
+struct LargeByteArray {
+  LargeByteArray() : len(0), ptr(NULLPTR) {}
+  LargeByteArray(uint64_t len, const uint8_t* ptr) : len(len), ptr(ptr) {}
```

Comment:
Ditto. 

All the binary values cannot exceed a length of INT32_MAX in the parquet specs, so reusing `ByteArray` is sufficient.